torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -446,18 +519,24 @@ def _init_from_local_shards(
             gathered_metadatas = [local_sharded_tensor_metadata]
 
         global_sharded_tensor_metadata = build_global_metadata(gathered_metadatas)
+        tensor_properties = global_sharded_tensor_metadata.tensor_properties
 
         # STEP 3: Validation done, create the actual ShardedTensor and populate fields
         # prepare initialization
-        sharded_tensor = cls.__new__(cls)
+        spec = shard_spec._infer_sharding_spec_from_shards_metadata(
+            global_sharded_tensor_metadata.shards_metadata
+        )
+        sharded_tensor = cls.__new__(cls,
```

Comment:
Can't we use `cls(...)` instead of `cls.__new__(...)`?

Reply:
I didn't check if using `cls(...)` will trigger `__init__` or not, ideally we don't want to trigger `__init__` here and that's why I specified explicitly, I will check this as a follow up and remove `__new__` if `cls(...)` directly works for us :)